lib/cosmos/tools/script_runner/script_runner_frame.rb

```diff
@@ -678,7 +678,7 @@ def post_line_instrumentation(filename, line_number)
     def exception_instrumentation(error, filename, line_number)
       if error.class == StopScript || error.class == SkipTestCase || !@use_instrumentation
         Kernel.raise error
-      else
+      elsif @@error != error
```

Comment:
I don't think this is doing what you want.  Two equivalent errors are considered equal by == (such as if you hit retry on a line that fails, the second exception would == the first).  I think you want to use:
```
elseif !@@error.eql?(error)
```

Reply:
Wow, my implementation totally broke the default case. Can't believe I didn't notice that. I think I tested this new change better in all cases it appears to do what we want.